group_backend/models/res_users.py

```diff
@@ -0,0 +1,23 @@
+from odoo import api, models
+
+
+class Users(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def has_group(self, group_ext_id):
+        """While ensuring a user is part of `base.group_user` this code will
+        try if user is in the `group_backend.group_backend` group to let access
+        to the odoo backend.
+
+        This code avoid to overwrite a lot of places in controllers from
+        different modules ('portal', 'web', 'base') with hardcoded statement
+        that check if user is part of `base.group_user` group.
+
+        As far `base.group_user` have a lot of default permission this
+        makes hard to maintain proper access right according your business.
+        """
+        res = super(Users, self).has_group(group_ext_id)
+        if not res and group_ext_id == "base.group_user":
```

Comment:
I think this is a smart hack avoiding a lot of mess in the code.

However it makes the behavior of this module very unpredictable. At anytime, the native code of Odoo could be changed to rely on `has_group` to grant additional access and then we would loose control again on what the users of that group could or could not do. And since this is touching security, this is one of the few things that might change within the same major version of Odoo.

It seems that `web` module only relies one time on `self.env.user.has_group('base.group_user')` and for `base` module there are 5 cases of usage (some might not be easy to override). And it might be acceptable to not override all cases, because overriding would only make sense if you want to give the permission which might not be needed for this very special user type. So overriding only the few necessary places could make sense.

I don't think `portal` module is necessary to take into consideration as it is not really "core" (it's not auto_install anymore in v14).

In my opinion, the code of this PR could:

- stay as it is if highlighting more the risk
  OR
- add the code to override wherever necessary

To highlight the risk:

- the content of this docstring can be added as a Limitation in the Readme
- a warning log could alert that we are bypassing the expected behavior of has_group.

```
if not res and group_ext_id == "base.group_user":
    has_group_backend = super(Users, self).has_group("group_backend.group_backend")
    if has_group_backend:
        logging.waning("Forcing has_group to return True for group_backend")
```

Reply:
Thanks a lot for your feed backs.

I've fist had a try to override the code in different parts but was no very confident as behavior would change according module dependency tree if portal or website are installed this would make difficult to predict which one will be called last (thinking `Home._login_redirect` or `Home.index` methods) and I don't want to depends on such modules.

So I think it's very good Idea to add a warning log and prevent against risk in README limitation.

I'll do that first.